src/mam/mod_mam.erl

```diff
@@ -144,6 +148,23 @@
 %% ----------------------------------------------------------------------
 %% API
 
+-spec get_personal_data(jid:user(), jid:server()) ->
+    [{gdpr:data_group(), gdpr:schema(), gdpr:entries()}].
+get_personal_data(Username, Server) ->
+    Schema = ["id", "from", "message"],
```

Comment:
it would be pretty hard to add any new fields, because the logic is in 10 files.